music21/environment.py

```diff
@@ -501,12 +501,20 @@ def getDefaultRootTempDir(self):
         '/var/folders/x5/rymq2tx16lqbpytwb1n_cc4c0000gn/T'
 
         >>> import os
+        >>> import sys
         >>> e = environment.Environment()
-        >>> e.getDefaultRootTempDir() == pathlib.Path(t) / 'music21'
+        >>> src_dir = e.getDefaultRootTempDir()
+        >>> tgt_dir= pathlib.Path(t) / 'music21'
+        >>> accessible = os.access(tgt_dir, os.R_OK) and os.access(tgt_dir, os.W_OK)
+        >>> if sys.platform.startswith('linux') and not accessible:\
+                tgt_dir = pathlib.Path(t) / f'music21-{os.getuid()}';
```

Comment:
This is too complex for a doctest, which is supposed to show users how to use the API. But I understand you wanted to doctor it so that it would pass on CI running Linux. Unfortunately we need to fake the doctest a little bit with #_DOCS_HIDE or #_DOCS_SHOW to hide and show lines to run or not run (see examples elsewhere in this file), and then move this test to a unit test. You can add a decorator for skipUnless so that your test only runs on Linux platforms.

Reply:
Got it. But I'm not familiar with doc tests and unit tests. It may take some time :P

Reply:
No worries! The good news is that the doctest passes with your feature without having to alter it, so I think we should just leave this doctest as is. I verified [here--not-real PR for testing only](https://github.com/jacobtylerwalls/music21/pull/29).

Then finding a way to test this works in a unit test -- probably better to mock the results of `os.access()` if possible than change read/write permissions of `tmp/`. Maybe you can look into [unittest mocking](https://docs.python.org/3/library/unittest.mock.html)?

Reply:
I took the other approach in 10f097274e of trying to fiddle with permissions in a unittest when you and I were working in this area last, but because I was trying to be cautious about fiddling with people's systems, the test doesn't run on CI, and so we didn't notice that it needs updating with your changes:

https://github.com/cuthbertLab/music21/blob/10f097274e82199132f50073d6cd89ba2ca85411/music21/environment.py#L1647

So I guess I'm open to both possibilities, you can try to fiddle with permissions, but honestly it's probably better to just mock. But we should update the test I just pointed to.

Reply:
As in, the test passes on my Mac because it's not Linux, but if this test ever runs on a Linux system where there is read/write access to tmp/ (apparently not on GitHub Actions) it will fail because `getpass.getuser()` != `os.getuid()`.

Reply:
I check [10f0972](https://github.com/cuthbertLab/music21/commit/10f097274e82199132f50073d6cd89ba2ca85411). I agree that it needs to be update. But I found these two part is used to deal with the same problem. It looks a little bit redundant. Should we integrate them into a single block? I mean remove the exception handling in `getTempFile()`, and only catch it in `getDefaultRootTempDir()`. Cause `getTempFile()` use `getDefaultRootTempDir()` to get the temp directory, we just need to make sure that `getDefaultRootTempDir()` provides a accessible and permission-allowed directory. Then we only need to maintain a single code if a new problem occurs in the future,

Reply:
Yes, that sounds great, let's simplify here and just have the permission checking in getDefaultRootTempDir().